src/components/common/Modal.tsx

```diff
@@ -0,0 +1,166 @@
+import React from 'react';
+import { createStyles, withStyles, WithStyles } from '@mui/styles';
+import {
+  Button,
+  Dialog,
+  DialogTitle,
+  DialogActions,
+  DialogContent,
+  IconButton,
+  CustomTheme,
+  Typography,
+} from '@mui/material';
+import CloseIcon from '@mui/icons-material/Close';
+import LoadingSpinner from 'components/common/LoadingSpinner';
+
+const styles = (theme: CustomTheme) =>
+  createStyles({
+    wrapper: {
+      padding: theme.spacing(4),
+      margin: theme.spacing(4),
+    },
+    dialogTitle: {
+      borderBottom: `1px solid ${theme.palette.divider}`,
+      margin: 0,
+      padding: theme.spacing(2),
+    },
+    closeButton: {
+      position: 'absolute',
+      right: theme.spacing(1),
+      top: theme.spacing(1),
+      color: theme.palette.grey[500],
+    },
+    dialogContent: {
+      margin: 0,
+      padding: `${theme.spacing(2)} !important`,
+    },
+    dialogInstructions: {
+      margin: 0,
+      padding: `${theme.spacing(2)} !important`,
+      whiteSpace: 'pre-wrap',
+    },
+    dialogActions: {
+      borderTop: `1px solid ${theme.palette.divider}`,
+      margin: 0,
+      padding: theme.spacing(1),
+    },
+    openButton: {
+      width: '100%',
+      border: 0,
+      justifyContent: 'left',
+      textTransform: 'none',
+      paddingTop: theme.spacing(1),
+      paddingBottom: theme.spacing(1),
+      '&:hover': {
+        border: 0,
+      },
+    },
+    openButtonIcon: {
+      marginRight: 5,
+      top: theme.spacing(1),
+    },
+    horizontalModalButton: {
+      fontSize: theme.typography.h6.fontSize,
+      color: theme.palette.primary.light,
+    },
+    overlaySpinner: {
+      opacity: 0.9,
+      position: 'absolute',
+      right: 0,
+      bottom: 0,
+      left: 0,
+      width: 'initial',
+      overflow: 'clip',
+      backgroundColor: theme.palette.common.white,
+      zIndex: 100,
+    },
+  });
+
+interface ModalProps extends WithStyles<typeof styles> {
+  modalText: string;
+  modalContent: string;
+  modalHeading: string;
+  isLoading: boolean;
+}
+
+type ModalState = {
+  open: boolean;
+};
+
+const initialState: ModalState = {
+  open: false,
+};
+
+export class Modal extends React.PureComponent<ModalProps, ModalState> {
+  constructor(props: ModalProps) {
+    super(props);
+    this.state = initialState;
+  }
```

Comment:
Rather than using a constructor and typing state, we can do this using a functional component. 
```suggestion
export const Modal = (props: ModalProps) => {
  const [open, setOpen] = useState(false)
```